scene/main/viewport.cpp

```diff
@@ -3467,17 +3468,62 @@ bool Viewport::is_using_xr() {
 	return use_xr;
 }
 
-void Viewport::set_scale_3d(float p_scale_3d) {
+void Viewport::set_scaling_3d_mode(Scaling3DMode p_scaling_3d_mode) {
+	if (scaling_3d_mode == p_scaling_3d_mode) {
+		return;
+	}
+
+	scaling_3d_mode = p_scaling_3d_mode;
+	RS::get_singleton()->viewport_set_scaling_3d_mode(viewport, (RS::ViewportScaling3DMode)(int)p_scaling_3d_mode);
+}
+
+Viewport::Scaling3DMode Viewport::get_scaling_3d_mode() const {
+	return scaling_3d_mode;
+}
+
+void Viewport::set_scaling_3d_quality(float p_scaling_3d_quality) {
 	// Clamp to reasonable values that are actually useful.
 	// Values above 2.0 don't serve a practical purpose since the viewport
 	// isn't displayed with mipmaps.
-	scale_3d = CLAMP(p_scale_3d, 0.1, 2.0);
+	scaling_3d_quality = CLAMP(p_scaling_3d_quality, 0.1, 2.0);
+
+	RS::get_singleton()->viewport_set_scaling_3d_quality(viewport, scaling_3d_quality);
+}
+
+float Viewport::get_scaling_3d_quality() const {
+	return scaling_3d_quality;
+}
+
+void Viewport::set_fsr_sharpness(float p_fsr_sharpness) {
+	if (fsr_sharpness == p_fsr_sharpness) {
+		return;
+	}
+
+	if (p_fsr_sharpness > 2.0f) {
+		p_fsr_sharpness = 2.0f;
+	} else if (p_fsr_sharpness < 0.0f) {
+		p_fsr_sharpness = 0.0f;
+	}
```

Comment:
I dont think we need to clamp this to ``2.0``. The FSR docs just say that values above 2 won't make a visible difference.